smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/GradleGenerator.kt

```diff
@@ -39,20 +45,57 @@ fun writeGradleBuild(
         }
     }
 
-    writer.withBlock("dependencies {", "}\n") {
-        write("implementation(kotlin(\"stdlib\"))")
-        // TODO - can we make kotlin dependencies not specify a version e.g. kotlin("kotlin-test")
-        // TODO - Kotlin MPP setup (pass through KotlinSettings) - maybe separate gradle writers
-        val orderedDependencies = dependencies.sortedWith(compareBy({ it.config }, { it.artifact }))
-        for (dependency in orderedDependencies) {
-            write("${dependency.config}(\"#L:#L:#L\")", dependency.group, dependency.artifact, dependency.version)
+    if (isKmp) { // In KMP projects the following generated code lives within a 'kotlin' block.
```

Comment:
I'm currently investigating if we can use gradle model types in a gradle writer, so for now this is pretty scrappy.